WordPress/Classes/ViewRelated/Media/MediaLibraryPickerDataSource.m

```diff
@@ -543,27 +557,38 @@ - (NSString *)name
     return NSLocalizedString(@"WordPress Media", @"Name for the WordPress Media Library");
 }
 
-- (WPMediaRequestID)imageWithSize:(CGSize)size completionHandler:(WPMediaImageBlock)completionHandler
+- (void)refreshImageMedia
 {
     NSString *entityName = NSStringFromClass([Media class]);
     NSFetchRequest *request = [NSFetchRequest fetchRequestWithEntityName:entityName];
     request.predicate = [MediaLibraryPickerDataSource predicateForFilter:self.filter blog:self.blog];
     NSSortDescriptor *sortDescriptor = [NSSortDescriptor sortDescriptorWithKey:@"creationDate" ascending:NO];
     request.sortDescriptors = @[sortDescriptor];
+    request.fetchLimit = 1;
     NSError *error;
     NSArray *mediaAssets = [[[ContextManager sharedInstance] mainContext] executeFetchRequest:request error:&error];
-    if (mediaAssets.count == 0)
-    {
-        if (completionHandler){
-            completionHandler(nil, nil);
-        }
-    }
     Media *media = [mediaAssets firstObject];
-    if (!media) {
+    self.imageMediaID = media.objectID;
+}
+
+- (WPMediaRequestID)imageWithSize:(CGSize)size completionHandler:(WPMediaImageBlock)completionHandler
+{
+    Media *media = nil;
+
+    if (self.imageMediaID == nil) {
+        [self refreshImageMedia];
+    }
+
+    if (self.imageMediaID != nil) {
+        media = [[[ContextManager sharedInstance] mainContext] existingObjectWithID:self.imageMediaID error:nil];
+    }
+
+    if (media == nil) {
         UIImage *placeholderImage = [UIImage imageNamed:@"WordPress-share"];
         completionHandler(placeholderImage, nil);
```

Comment:
I noticed that, I also noticed that the `completionHandler` is called twice when there is no media assets–this function doesn't return after the first `completionHandler` call. The first call does a nil-check, but the second doesn't. This means the `completionHandler` shouldn't be nil in reality.

I think, given this function is used to get the image of the media asset, the call site is supposed to pass an `completionHandler`, otherwise what would they call this function for? This is the main reason why I didn't do a nil-check here—passing a nil `completionHandler` is a programming error.

Reply:
👌